rust_src/src/list.rs

```diff
@@ -0,0 +1,37 @@
+use std::os::raw::c_char;
```

Comment:
Since lists are just collections of cons cells, I propose that we move these functions into cons.rs. We could move `Fatom` into cons.rs too.